core/src/main/java/org/zenoss/zep/zing/ZingEvent.java

```diff
@@ -212,25 +217,51 @@ public Event toZingEventProto() {
             //--------------------------------------------
 
             // Status
-            if (!ZingUtils.isNullOrEmpty(this.status)){
+            if (!ZingUtils.isNullOrEmpty(this.status)) {
                 // default: Status.STATUS_DEFAULT;
                 b.putMetadata( ZingConstants.STATUS_KEY, ZingUtils.getScalarArray(this.status));
-                b.setStatus(statusMap.get(this.status));
 
-                if (this.status == "ACKNOWLEDGED") {
+                // Ensure status gets mapped faithfully, else log:
+                if (!statusMap.containsKey(this.status)) {
+                    logger.warn("No key in statusMap for status: {}", this.status);
+
+                    // Logging for debugging only: Remove after done...
+                    for (String key: statusMap.keySet()) {
+                        logger.debug("Available Key: {}", key);
+                    };
+                    logger.debug("Setting default status for missing status key: {}", this.status);
+                    b.setStatus(Status.STATUS_DEFAULT);
+                } else {
+                    b.setStatus(statusMap.get(this.status));
+                }
+
+                // For Acknowledged status, set the corresponding property
+                if (this.status == "STATUS_ACKNOWLEDGED") {
```

Comment:
this compares the addresses; within parens should say "STATUS_ACKNOWLEDGED".equals(this.status)